### DISCUSSION TITLE
[1.20.1] Incompatibilities/Compatibilities of the Connector

### DISCUSSION THREAD

**Discussion by Starexify** [Sinytra/Connector#180] [General]
## Info
I wanted to make this list of mods that doesn't work together with the Connector, for the reason that if someone finds incompatibilites and doesnt know which mod does that, to come here and find out which mods are incompatible. Some fabric mods are incompatible with other fabric mods aswell!

# Incompatible Mods
| 📦Mod Names | 📢 Status | ✍️  Notes |
| --- | --- | --- |
| BedrockIfy | ❌ | |
| Charmonium | ❌ | Euphonium replaces it |
| Craftify | ❌ | |
| Debugify | ❌ | |
| Early Loading Screen | ❌ | |
| Pet Armor | ❌ | Better Nether or other mod doesnt work with it |
| Sodium | ❌ | Use Xenon or Embeddium better |
| The Backrooms | ❌ | |

# Compatible Mods
| 📦Mod Names | 📢 Status | ✍️  Notes |
| --- | --- | --- |
| Adaptive Tooltips | ✔️ | Requires [YetAnotherConfigLib](https://modrinth.com/mod/yacl) |
| Alloy Forgery | ✔️ | |
| Amethyst Imbuement | ✔️ | |
| BedrockWaters | ✔️ | Crash with Embeddium/Xenon |
| Better End | ✔️ | |
| Better Nether | ✔️ | |
| Better Nether Map | ✔️ | |
| Better Statistics Screen | ✔️ | |
| Biome Fog | ✔️ | |
| Biome Moss | ✔️ | |
| Blahaj (Fabric) | ✔️ | |
| Bosses of Mass Destruction | ✔️ | |
| CameraOverhaul | ✔️ | |
| Cave Dust | ✔️ | |
| Chat Animation | ✔️ | Chat Impressive Animation exists tho |
| Chat Sounds | ✔️ | |
| Chunks fade in | ✔️ | Needs config file |
| Compact Chat | ✔️ | |
| Continuity | ✔️ | |
| Detail Armor Bar | ✔️ | |
| Disenchanter | ✔️ | |
| Dynamic Surroundings: Fabric Edition | ✔️ | |
| Euphonium | ✔️ | |
| Faded Widgets | ✔️ | |
| Kroko’s Elementary Staffs | ✔️ | |
| Enhanced Block Entities | ✔️ | |
| Explosive Enhancement | ✔️ | The chests glitch but if you disable enhanced chests they won't glitch anymore |
| Hybrid Aquatic | ✔️ | |
| ImmersiveThunder | ✔️ | |
| ItemSwapper | ✔️ | |
| Jump Sounds | ✔️ | |
| MC Dungeons Enchanting | ✔️ | |
| Mod-erate Loading Screen | ✔️ | |
| Mythic Metals | ✔️ | |
| Numeral Ping | ✔️ | |
| Nvidium/Revidium | ✔️ | Look down in the Nvidium Parity section |
| oωo (owo-lib) | ✔️ | |
| Reimagined World Menu | ✔️ | |
| Show Me What You Got | ✔️ | |
| Smooth Particles | ✔️ | |
| Smooth Swapping | ✔️ | |
| SuperBetterGrass | ✔️ | |
| Traverse | ✔️ | |
| Trinkets | ✔️ | |
| Universal Ores | ✔️ | |
| Vectorientation | ✔️ | |
| Wakes | ✔️ | |

# Nvidium parity beta37+
1. Put these 2 config files in the folder:
[connector.json](https://github.com/Sinytra/Connector/files/14500807/connector.json)
[fabric_loader_dependencies.json](https://github.com/Sinytra/Connector/files/14129116/fabric_loader_dependencies.json)

If you use ChunksFadeIn just use this instead of the 2nd config file:
[fabric_loader_dependencies.json](https://github.com/Sinytra/Connector/files/14129123/fabric_loader_dependencies.json)

# Nvidium parity under beta36
1. Put these 2 config files in the folder:
[connector_global_mod_aliases.json](https://github.com/Sinytra/Connector/files/14129106/connector_global_mod_aliases.json)
[fabric_loader_dependencies.json](https://github.com/Sinytra/Connector/files/14129116/fabric_loader_dependencies.json)

If you have any mods that are incompatible, leave them below. I'll try to add them all!

**Comment by ofmeteoriteh** [comment]
Chunks fade in,it seems to be possible to run by adding a config file in the config folder.
[fabric_loader_dependencies.json](https://github.com/Sinytra/Connector/files/14001523/fabric_loader_dependencies.json)

**Reply by Starexify** [reply]
Thanks for the info I changed now👍

**Comment by ofmeteoriteh** [comment]
I just found out that this configuration file needs to be loaded in the game and needs to be pre-loaded, add it to the config folder, and then restart it to use Chunks fade in normally.
If it is not loaded, it may cause the game to crash.

**Comment by Danieloxor** [comment]
Nvidium works just fine alongside Embeddium with some dependency config tweaking. i found the solution in this [reddit comment](https://www.reddit.com/r/allthemods/comments/199sm7i/comment/kihjyak/?utm_source=share&utm_medium=web2x&context=3)

I used the same way for Indium

**Reply by kakashi1464** [reply]
Use lazurite instead of indium

**Reply by Starexify** [reply]
It doesnt work sadly, nor revidium. The blocks are just invisible. So sadly no nvidium compat

**Reply by Danieloxor** [reply]
> It doesnt work sadly, nor revidium. The blocks are just invisible. So sadly no nvidium compat

Edit: Found the issue, Embeddium 0.3.0 breaks nvidium. Use 0.2.18

### Yes it does. at least for me.

![image](https://github.com/Sinytra/Connector/assets/113946717/4aeacafe-cb60-4bf3-bf5e-0109e8528526)

Here i am with Embeddium 0.2.18, Lazurite, ImmediatlyFast, and ModernFix. Running on NeoForged 47.1.96
But it's about.. idk 40% the performance of Nvidium on Quilt. but i tested nvidium on quilt with many others performance mods, and here i have some big mods installed already so comparison is unfair. but still i think on forge i got less fps, as expected 

My configs:

 `connector_global_mod_aliases.json`:
```
{
  "version": 1,
  "aliases": {
  "cloth_config": "cloth-config2",
  "playeranimator": "player-animator",
  "embeddium": ["sodium"]
  }
}
```
 `fabric_loader_dependencies.json`:
```
{
    "version": 1,
    "overrides":{
        "nvidium":{
        "-depends":{
        "sodium": "IGNORED"
            },
        "+depends":{
        "sodium": "*"
            }
        }
    }
}
```

**Reply by Ivy31415** [reply]
@Starexify try turning off translucent face culling, i did that and it fixed all of the issues with blocks not rendering

**Reply by Starexify** [reply]
I will try

**Reply by Starexify** [reply]
I can't find the option "Translucent Face Culling" anywhere

**Reply by Ivy31415** [reply]
should be options>video settings>advanced
just checked it's called translucent face sorting

**Reply by Starexify** [reply]
Which embeddium and nvidium versions ??
Nvm I found out nvidium updated and I didnt update it lol so it works now thanks tho

**Comment by Danieloxor** [comment]
> I used the latest embeddium thats why it broke didnt work thx

Yup xd when I tested 0.2.18 was latest. So you can edit the compatibility table with note about needing to edit config files and using specific versions

**Reply by Starexify** [reply]
I tried again and it still doesnt work, I used the

```connector_global_mod_aliases.json:```
```
{
  "version": 1,
  "aliases": {
  "cloth_config": "cloth-config2",
  "playeranimator": "player-animator",
  "embeddium": ["sodium"]
  }
}
```

```fabric_loader_dependencies.json:```
```
{
    "version": 1,
    "overrides":{
        "nvidium":{
        "-depends":{
        "sodium": "IGNORED"
            },
        "+depends":{
        "sodium": "*"
            }
        }
    }
}
```
and only used these mods on latest version Neoforge 1.20.1:
![image](https://github.com/Sinytra/Connector/assets/80699736/cae7477f-e63b-4a86-b98a-9afb09e5d390)
It gives me this crash: https://pastebin.com/2gxxzjgC

Edit: Found out its not nvidium that breaks the launching and immediatelyfast

**Reply by Danieloxor** [reply]
It should work even without ImmediatelyFast and Lazurite.

Try without ImmediatelyFast, im having ImmediatelyFast-Forge-1.2.8+1.20.4. maybe NeoForge version causing a crash for you?

Edit: here are minimum mods i used to run Nvidium on latest forge 47.2.20:
```
Mods:
  [🖿]  (folder)
  [✔] Connector-1.0.0-beta.35+1.20.1
  [✔] ConnectorExtras-1.9.2+1.20.1
  [✔] embeddium-0.2.18+mc1.20.1
  [✔] fabric-api-0.91.0+1.10.8+1.20.1
  [✔] nvidium-0.2.5-beta
  ```
  and of course modified configs
  
  Edit: 

> Edit: Found out its not nvidium that breaks the launching and immediatelyfast

So it needs to run with Forge version of immediatelyfast instead of Neoforge. Weird but ok

**Reply by Starexify** [reply]
On NeoForge it works with the forge version of ImmediatelyFast. But I think one of the fabric mods I have crashes it, I will try these days to see which crashes with nvidium and note everything

**Reply by Danieloxor** [reply]
Can you share with me your mods? maybe i encountered a problem with one of them while i had installed nvidium

**Reply by Danieloxor** [reply]
Good to know. i downloaded TexTrue before newest version was released. 

Off topic - "Iron's Spells and Spellboks" don't work with fabric GeckoLib. Crashes when rendering 3D models.

**Comment by lilmayu** [comment]
Pipez's Item Pipe and Tech Reborn's machines (inventory blocks in general) incompatibility - they do not connect / transfer items.

![obrazek](https://github.com/Sinytra/Connector/assets/52459400/76ed847e-a5c1-4913-ab71-ebf33d0d5cf4)

However, Item Pipes shows a connection to Tech Reborn's Storage Units, but they do not transfer any items.

Not sure how it would be fixed, thought. I would imagine that it would connect to any block entities that extend somekind of Inventory interface.

Note that Fluid and Energy pipes works, thanks to Connector Extras.

Connector 1.0.0-beta.39+1.20.1
Pipez 1.2.6
TechReborn 5.8.7

**Reply by lilmayu** [reply]
Well, after hours and hours of learning, browsing Tech Reborn's code, learning Fabric's transaction API and Forge's way of handling items, I have came up with this `IItemHandler` that can be used as `ForgeCapabilities.ITEM_HANDLER`. Works pretty well for me, atleast. Not sure how bug-prone it is.

Also, I do not know how I would get limit of the stack size for specified slot in the container, since vanilla's container does not care about per-slot max stack size to my knowledge.

Not sure if usable, but sending it here regardless.

```java
/**
 * A Forge IItemHandler that wraps a Fabric Storage
 *
 * @author <a href="https://mayuna.dev">Mayuna</a>
 */
@SuppressWarnings("UnstableApiUsage")
public class FabricItemStorage implements IItemHandler {

    private final Storage<ItemVariant> storage;
    private final Container container;

    /**
     * Create a new FabricItemStorage
     *
     * @param storage   The storage to use
     * @param container The container to use
     */
    public FabricItemStorage(Storage<ItemVariant> storage, Container container) {
        this.storage = storage;
        this.container = container;

    }

    @Override
    public int getSlots() {
        return container.getContainerSize(); // Fabric's Storage API does not have a concept of slots, so we return the size of the container
    }

    @Override
    public @NotNull ItemStack getStackInSlot(int i) {
        return container.getItem(i); // Fabric's Storage API does not have a concept of slots, so we return the ItemStack in the specified slot from the container
    }

    @Override
    public @NotNull ItemStack insertItem(int amount, @NotNull ItemStack itemStack, boolean simulate) {
        if (itemStack.isEmpty()) {
            return itemStack; // Uhh... what?
        }

        // Open a transaction to insert the item into the storage
        try (Transaction transaction = Transaction.openOuter()) {
            // Insert the item into the storage
            long inserted = storage.insert(ItemVariant.of(itemStack), itemStack.getCount(), transaction);

            // If the transaction was simulated or no items were inserted, return the original stack
            if (simulate || inserted == 0) {
                return itemStack;
            }

            // If the transaction was not simulated, return a copy of the stack with the amount of items that were not inserted
            transaction.commit();
            return ItemHandlerHelper.copyStackWithSize(itemStack, (int) (itemStack.getCount() - inserted));

        }
    }

    @Override
    public @NotNull ItemStack extractItem(int slot, int amount, boolean simulate) {
        ItemStack stack = container.getItem(slot); // Get the ItemStack in the specified slot

        // If the ItemStack is empty, return an empty stack
        if (stack.isEmpty()) {
            return ItemStack.EMPTY;
        }

        // Open a transaction to extract the item from the storage
        try (Transaction transaction = Transaction.openOuter()) {
            long extracted = storage.extract(ItemVariant.of(stack), amount, transaction);

            // If the transaction was simulated or no items were extracted, return an empty stack
            if (extracted == 0) {
                return ItemStack.EMPTY;
            }

            // If the transaction was not simulated, commit the transaction
            if (!simulate) {
                transaction.commit();
            }

            // Return a copy of the stack with the amount of items that were extracted
            return ItemHandlerHelper.copyStackWithSize(stack, (int) extracted);
        }
    }

    @Override
    public int getSlotLimit(int slot) {
        // TODO: Get max size of stack in specific slot.. Dunno how
        return container.getMaxStackSize(); // Usually 64
    }

    @Override
    public boolean isItemValid(int slotIndex, @NotNull ItemStack itemStack) {
        return container.canPlaceItem(slotIndex, itemStack);
    }
}
```

**Comment by AdeptusAutumn** [comment]
[NEEPMeat](https://modrinth.com/mod/neepmeat) does not load and will not even produce an crash report. Have a [latest.log](https://mclo.gs/Q1J8fcb). Install contains only the barebones and dependency. 
Forge 47.1.3
fabric-api-0.92.1+1.11.8+1.20.1.jar (Forgified Fabric)
Connector-1.0.0-beta.42+1.20.1
ConnectorExtras-1.10.0+1.20.1
geckolib-forge-1.20.1-4.4.4
neepmeat-0.2.24-beta+1.20.1

**Reply by kakashi1464** [reply]
Use fabric geckolib

**Comment by LYNZEJL** [comment]
Does ars noveu work With sinytra

**Reply by kakashi1464** [reply]
Yes

**Comment by electikos** [comment]
does someone know about a way to make synitra compatible with embeddium ?

**Reply by kakashi1464** [reply]
Embeddium is compatible with connector

**Reply by electikos** [reply]
it wasn't for me because of TLauncher that make a update later.

**Comment by Aarav-0001** [comment]
I was playing with sinytra connector and when i wore a armor piece my game crashed and showed up some armor stand issue , the mods that might have caused the issue are better end , better nether and epic fight i think these mods might be incompatabile but i tried this like a months ago so i dont know if this issue still comes up i am going to check that right after this. I can provide u with logs and crash report if that helps.

**Reply by Aarav-0001** [reply]
i tried to replicate it but this time minecraft just went unresponsive and didnt respond so i closed it from jdk

**Reply by Aarav-0001** [reply]
i found a photo of the same error that i saved 
![image_2024-11-15_201922490](https://github.com/user-attachments/assets/f66215f1-26f9-4546-be85-34c519b15ea4)
if somebody tries to replicate this their game might crash but they might not get this exact crash log but i think it is the same problem

**Reply by jediultras** [reply]
Yeah, exact problem for me, I tried everything. The only thing that I can think of is trying to downgrade epic fight to a version that might not crash, but I don't know how much I need to go back. I also wanted to try to post this issue on their GitHub issues page and see if they can resolve this problem

**Comment by Mishantaaa** [comment]
> ## Info
> I wanted to make this list of mods that doesn't work together with the Connector, for the reason that if someone finds incompatibilites and doesnt know which mod does that, to come here and find out which mods are incompatible. Some fabric mods are incompatible with other fabric mods aswell!
> 
> # Incompatible Mods
> 📦Mod Names	📢 Status	✍️ Notes
> BedrockIfy	❌	
> Charmonium	❌	Euphonium replaces it
> Craftify	❌	
> Debugify	❌	
> Early Loading Screen	❌	
> Pet Armor	❌	Better Nether or other mod doesnt work with it
> Sodium	❌	Use Xenon or Embeddium better
> The Backrooms	❌	
> # Compatible Mods
> 📦Mod Names	📢 Status	✍️ Notes
> Adaptive Tooltips	✔️	Requires [YetAnotherConfigLib](https://modrinth.com/mod/yacl)
> Alloy Forgery	✔️	
> Amethyst Imbuement	✔️	
> BedrockWaters	✔️	Crash with Embeddium/Xenon
> Better End	✔️	
> Better Nether	✔️	
> Better Nether Map	✔️	
> Better Statistics Screen	✔️	
> Biome Fog	✔️	
> Biome Moss	✔️	
> Blahaj (Fabric)	✔️	
> Bosses of Mass Destruction	✔️	
> CameraOverhaul	✔️	
> Cave Dust	✔️	
> Chat Animation	✔️	Chat Impressive Animation exists tho
> Chat Sounds	✔️	
> Chunks fade in	✔️	Needs config file
> Compact Chat	✔️	
> Continuity	✔️	
> Detail Armor Bar	✔️	
> Disenchanter	✔️	
> Dynamic Surroundings: Fabric Edition	✔️	
> Euphonium	✔️	
> Faded Widgets	✔️	
> Kroko’s Elementary Staffs	✔️	
> Enhanced Block Entities	✔️	
> Explosive Enhancement	✔️	The chests glitch but if you disable enhanced chests they won't glitch anymore
> Hybrid Aquatic	✔️	
> ImmersiveThunder	✔️	
> ItemSwapper	✔️	
> Jump Sounds	✔️	
> MC Dungeons Enchanting	✔️	
> Mod-erate Loading Screen	✔️	
> Mythic Metals	✔️	
> Numeral Ping	✔️	
> Nvidium/Revidium	✔️	Look down in the Nvidium Parity section
> oωo (owo-lib)	✔️	
> Reimagined World Menu	✔️	
> Show Me What You Got	✔️	
> Smooth Particles	✔️	
> Smooth Swapping	✔️	
> SuperBetterGrass	✔️	
> Traverse	✔️	
> Trinkets	✔️	
> Universal Ores	✔️	
> Vectorientation	✔️	
> Wakes	✔️	
> # Nvidium parity beta37+
> 1. Put these 2 config files in the folder:
>    [connector.json](https://github.com/Sinytra/Connector/files/14500807/connector.json)
>    [fabric_loader_dependencies.json](https://github.com/Sinytra/Connector/files/14129116/fabric_loader_dependencies.json)
> 
> If you use ChunksFadeIn just use this instead of the 2nd config file: [fabric_loader_dependencies.json](https://github.com/Sinytra/Connector/files/14129123/fabric_loader_dependencies.json)
> 
> # Nvidium parity under beta36
> 1. Put these 2 config files in the folder:
>    [connector_global_mod_aliases.json](https://github.com/Sinytra/Connector/files/14129106/connector_global_mod_aliases.json)
>    [fabric_loader_dependencies.json](https://github.com/Sinytra/Connector/files/14129116/fabric_loader_dependencies.json)
> 
> If you have any mods that are incompatible, leave them below. I'll try to add them all!

Why Origins is not working?

**Reply by Starexify** [reply]
I left this page since I stopped playing modded for a time and started modding instead (also no longer play on 1.20.1 so I won't update this any time sooner) but maybe someone else can help 👍

**Comment by Just-ElC** [comment]
Debugify is now compatibility

**Reply by kocane** [reply]
Do we need to disable anything in the settings of Debugify, or can we just install it?

**Reply by Just-ElC** [reply]
> Do we need to disable anything in the settings of Debugify, or can we just install it?

Just install it

**Comment by TernaryAlpaca** [comment]
origins and better nether do not work

**Comment by electikos** [comment]
there is a incompatibility with the mod '' defile '' from doctor4t (all the dependencies needed are: ratatouie and satin)

**Reply by AbhXD** [reply]
other mods of his dont work either. i tried arsenal and that didnt work

**Reply by electikos** [reply]
I know, it's because of the mixin : sinnytra don't let defile (or other mods) access ratatouille.

**Reply by electikos** [reply]
and there are no traduction (fabric to forge) of some mixin.

**Comment by GelatinousGoth** [comment]
It also seems to be incompatible with the newest 3-4 1.20.1 versions of the Effective mod. 
Causes an error on mod loading, explaining that it fails to get something from a github.

**Comment by Ex0r3ck** [comment]
VulkanMod is incompatible

**Comment by Exerra** [comment]
ClickSigns 1.0.4 doesn't work, however 1.0.2 and 1.0.3 do work.

**Comment by KermitEmperor** [comment]
Better Fog doesn't work completely, fog effects seem to be ignored when Sinytra is present.

**Reply by ReadZead** [reply]
could just be an issue like some other mods where for some reason with sinytra is doesn't create the necessary config files, best help I can give you is to check if you have the mod works fine without sinytra and to see if there are any issues with the mod reading the config file with sinytra reinstalled

**Comment by AETTEA10** [comment]
Hi, I am getting an crash before Minecraft boots and in the Modrinth logs it says BetterNether is responsible. Is there a way to fix this or is it just not compatible? Thanks.

**Reply by AETTEA10** [reply]
and here's the first bit of the crash logs:

net.minecraftforge.fml.loading.EarlyLoadingException: Could not execute entrypoint stage 'main' due to errors, provided by 'betternether' at 'org.betterx.betternether.BetterNether'!
at org.sinytra.connector.loader.ConnectorEarlyLoader.createLoadingException(ConnectorEarlyLoader.java:81) ~[Connector-1.0.0-beta.46+1.20.1.jar%23256!/:1.0.0-beta.46+1.20.1] {}
at org.sinytra.connector.loader.ConnectorEarlyLoader.createGenericLoadingException(ConnectorEarlyLoader.java:77) ~[Connector-1.0.0-beta.46+1.20.1.jar%23256!/:1.0.0-beta.46+1.20.1] {}
at org.sinytra.connector.mod.ConnectorLoader.load(ConnectorLoader.java:55) ~[Connector-1.0.0-beta.46+1.20.1-mod.jar%23260!/:1.0.0-beta.46+1.20.1] {re:mixin,re:classloading}
at net.minecraft.client.Minecraft.handler$cnb001$connectormod$earlyInit(Minecraft.java:21424) ~[client-1.20.1-20230612.114412-srg.jar%23358!/:?] {re:mixin,pl:accesstransformer:B,pl:connector_pre_launch:A,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:A,pl:connector_pre_launch:A,pl:runtimedistcleaner:A}
at net.minecraft.client.Minecraft.<init>(Minecraft.java:424) ~[client-1.20.1-20230612.114412-srg.jar%23358!/:?] {re:mixin,pl:accesstransformer:B,pl:connector_pre_launch:A,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:A,pl:connector_pre_launch:A,pl:runtimedistcleaner:A}
at net.minecraft.client.main.Main.main(Main.java:182) ~[1.20.1-47.4.9.jar:?] {re:mixin,pl:connector_pre_launch:A,pl:runtimedistcleaner:A,re:classloading,pl:mixin:APP:smoothboot.mixins.json:client.MainMixin from mod smoothboot,pl:mixin:APP:mixins.emibridge.json:MainEntrypointMixin from mod connectorextras_emi_bridge,pl:mixin:A,pl:connector_pre_launch:A,pl:runtimedistcleaner:A}
at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}

**Reply by ReadZead** [reply]
do you also have connector extras installed? if not install it and try again and for safety, delete the betternether.json file from your config folder if present

**Comment by AhmedTheGreatest** [comment]
Can I use the mod Auto Lectern with it? https://modrinth.com/mod/auto-lectern For version 1.20.1.

**Comment by AhmedTheGreatest** [comment]
If anyone has any idea that would be great help!

**Comment by Zhincore** [comment]
Raknetify seems to not work (again?)

```
@WrapWithCondition annotation on stopTimeoutPlayersOnKeepAlive could not find any targets matching 'method_12082' in net.minecraft.server.network.ServerGamePacketListenerImpl. Using refmap raknetify-fabric-refmap.json [PREINJECT Applicator Phase -> raknetify-fabric.mixins.json:server.MixinServerPlayNetworkHandler1_20_1 from mod raknetify -> Prepare Injections ->  -> wrapWithCondition$iaf000$raknetify$stopTimeoutPlayersOnKeepAlive(Lnet/minecraft/server/network/ServerGamePacketListenerImpl;Lnet/minecraft/network/chat/Component;)Z -> Parse]
``` 
(a lot more mixins don't succeed either)

**Comment by Yomagn-Tho** [comment]
Cardinal Components and Fabric Language Kotlin are both compatible, 
and dependent on HexConnect from Modrinth alongside either Sinytra Connector Extras or Sinytra Connector Extras with VS2 Support, 
so is Hex Ways (1.20 fork) for 1.20.1 also from Modrinth. 
So far no issues in a pack containing Embedium, Valkyrian Skies, Clockwork, ModernFix, Immersive Portals (for Forge), BetterEnd, BCLib, Hexcasting, Hexxy Skies, Hexal, MoreIotas, Create, Create Interactive, and other impactful mods. 

Immersive Portals for Fabric is incompatible, but the Forge version causes no issues.
The Corners Renewed a fork of The Corners for Fabric is unfortunately incompatible but with effort can get all the way to attempting to load or make a world before crashing the game so partial compatibility.
Eden Ring Reforked is currently incompatible.
Bewitchment is likewise incompatible.

**Reply by O-M-S1** [reply]
can you give me the full modlist because embeddium and hexal do not work together for me when i join a server
